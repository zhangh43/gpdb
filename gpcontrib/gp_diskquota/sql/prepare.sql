create extension diskquota;
select pg_sleep(1);
\! pg_ctl -D /tmp/pg_diskquota_test/data reload
\! cp data/csmall.txt /tmp/csmall.txt
select pg_sleep(5);

-- prepare a schema that has reached quota limit
create schema badquota;
select diskquota.set_schema_quota('badquota', '1 MB');
create role testbody;
create table badquota.t1(i int);
alter table badquota.t1 owner to testbody;
insert into badquota.t1 select generate_series(0, 50000);
select pg_sleep(5);
insert into badquota.t1 select generate_series(0, 10);
