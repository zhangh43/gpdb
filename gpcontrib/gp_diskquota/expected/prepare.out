create extension diskquota;
select pg_sleep(1);
 pg_sleep 
----------
 
(1 row)

\! pg_ctl -D /tmp/pg_diskquota_test/data reload
server signaled
\! cp data/csmall.txt /tmp/csmall.txt
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- prepare a schema that has reached quota limit
create schema badquota;
select diskquota.set_schema_quota('badquota', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

create role testbody;
create table badquota.t1(i int);
alter table badquota.t1 owner to testbody;
insert into badquota.t1 select generate_series(0, 50000);
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

insert into badquota.t1 select generate_series(0, 10);
ERROR:  schema's disk space quota exceeded with name:badquota
